src/test/java/com/wikia/webdriver/elements/oasis/components/navigation/UploadVideo.java

```diff
@@ -0,0 +1,44 @@
+package com.wikia.webdriver.elements.oasis.components.navigation;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+
+public class UploadVideo extends BasePageObject {
+
+  @FindBy(xpath = "//span[text()='Explore']/../../..")
+  private WebElement exploreButton;
+
+  @FindBy(css = ".wds-button.addVideo")
+  private WebElement addVideo;
+
+  @FindBy(css = "#VideoEmbedUrl")
+  private WebElement videoTextBox;
+
+  @FindBy(css = "#VideoEmbedUrlSubmit")
+  private WebElement submitVideo;
+
+  public UploadVideo exploreButton() {
+    new Actions(driver).moveToElement(exploreButton).perform();
+    return this;
+  }
+
+
+  public UploadVideo clickAddVideo() {
+    new Actions(driver).click(addVideo).perform();
+    return this;
+
+  }
+
+  public UploadVideo addVideo(String videoUrl) {
```

Comment:
pretty misleading what the difference between `clickAddVideo()` and `addVideo` methods. any chance to make them more descriptive?